resource-aggregate/events/deviceMetadataSnapshotTaken.go

```diff
@@ -93,8 +93,22 @@ func (e *DeviceMetadataSnapshotTaken) HandleDeviceMetadataSnapshotTaken(ctx cont
 	return nil
 }
 
-func (e *DeviceMetadataSnapshotTaken) HandleDeviceMetadataUpdatePending(ctx context.Context, updatePending *DeviceMetadataUpdatePending) error {
+func (e *DeviceMetadataSnapshotTaken) HandleDeviceMetadataUpdatePending(ctx context.Context, updatePending *DeviceMetadataUpdatePending, isCmd bool) error {
+	now := time.Now()
+	if !updatePending.IsValid(now) {
```

Comment:
Aggregate shall not validate commands. Command validation is done in a previous layer. It doesn't belong here.

Reply:
It's bellows here because, event can be expired during processing eventstore.